warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java

```diff
@@ -161,9 +161,10 @@ public void next() throws IOException {
                 ignore |= uidInfo.second();
                 
                 // Three things must be true in order to return a document specific range
-                // 1. The uids stored in the index must have the ignore flag set to FALSE
-                // 2. The option to collapse all document ranges must be false.
+                // 1. The uids stored in the index must NOT have the ignore flag set to FALSE
+                // 2. The option to collapse all document ranges cannot be set.
                 // 3. The document count must be under the document threshold.
+                // If any of those is false, do not add the uids.
```

Comment:
The wealth of double and even triple negatives is awesome.  Now the comment on line 163 and line 167 are in direct conflict with each other.  It seems to me that neither the comment nor the code was correct.  My reasoning is as follows:
1) If the uids stored in the index have the ignore flag set to FALSE, then there are no uids to add, hence ignore MUST be false if we are to add the UIDs, regardless of collapseUids state.
2) I agree that either the collapseUids should be false OR the count is less than the threshold in order to add the uids.
Hence I believe the correct comment and comparison should be:
```
// Two things must be true in order to return a document specific range:
// 1) The ignore flag must be false
// 2) Either the collapse flag is false, or the document count is under the document threshold
if (!ignore && (!collapseUids || (count <= collapseUidsThreshold))) {
```

Reply:
Good call, I'll make that update.